Feature new property

jListo opened the pull request:
Added: New Property Form, Navigation, API Call and Page.

Commit:
Base for Adding a New Property

The start of functionality to add a new property. IT IS CURRENTLY
NON-FUNCTIONAL.

Commit:
Test Parameter Changed

Changed the test parameter so the page would display correctly. STILL
ONLY PARTIALLY FUNCTIONAL

Commit:
Restructure and Bug Fixing

Restructured aspects of the code to be more appropriate and the feature
is now functional.

nirrek commented on scripts/components/NewPropertyForm.jsx:
```diff
@@ -0,0 +1,155 @@
+var Paper = mui.Paper;
+
+var NewPropertyForm = React.createClass({
+getInitialState() {
```
Indentation needs fixing

nirrek commented on scripts/components/NewPropertyForm.jsx:
```diff
@@ -0,0 +1,155 @@
+            multiLine={true}
+            onChange={this.onChange.bind(this, 'tenantEmail')}
+            floatingLabelText="Tenant Email" />
+          <TextField
```
This needs to be a `<input type='file' onChange={this.onFileSelected}/>` if you want a data URL encoding of a selected image.

nirrek commented on scripts/components/NewPropertyForm.jsx:
```diff
@@ -0,0 +1,155 @@
+            type="submit"
+            label="Add New Property"
+            primary={true}
+            backgroundColor="#2ECC71"
```
backgroundColor property not needed now that we have a material ui palette being applied to everything.

Commit:
Indentation, Style and API Changes

Indentation and style changed as per pull request. API call changed to
match the new endpoint.

nirrek committed:
(NewPropertyForm) add image upload and snackbar success notification

nirrek committed:
Merge branch 'master' into feature-new-property

nirrek committed:
Merge branch 'master' into feature-new-property

nirrek committed:
(Login) fix null pointer bug

nirrek committed:
(PropertyDetails) show correct address
